Add script2 to the `de-go-analysis` module  (3/5) (#236)
stjude-dnb-binfcore/sc-rna-seq-snap: analyses/de-go-analysis/02-volcano-plots.Rmd

    font-size: 20px;}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
  background-color: var(--DNB_BINF_Core_color);}
</style>

<a href="https://wiki.stjude.org/display/CAB">

</a>

\pagebreak

<div class="reporthead"><br/>
**PI: `r params$PI_NAME`**  
**Project: `r params$PROJECT_NAME`**  
Task: `r params$TASK_ID`  
Project Lead(s): `r params$PROJECT_LEAD_NAME`  
Department: `r params$DEPARTMENT`  

<br />  

DNB Bioinformatics Core Analysis Team: 
<br />  

>**Lead Analyst(s): `r params$LEAD_ANALYSTS`**  
>Group Lead: `r params$GROUP_LEAD`  
<br />
>**Contact E-mail:** `r params$CONTACT_EMAIL`  
>**DNB Bioinformatics Core Pipeline:** `r params$PIPELINE`  

Date started: `r params$START_DATE`  
Date completed:  `r params$COMPLETION_DATE`  
Report generated: `r format(Sys.time(), '%H:%M:%S %Z %m/%d/%Y')` \

Reviewed by: _____________________   Date: ____________ \
</div>
\pagebreak
  
# Information about this notebook

This notebook generates volcano plots to visualize and summarize DE patterns.

A volcano plot visualizes two key metrics for every gene:

- log₂ fold change (log2FC) — the magnitude and direction of expression change

> - statistical significance — usually represented as –log₁₀(adjusted p-value)

Reviewer: Since the module is using unadjusted p-value, you might consider changing this to avoid confusion, e.g.

```suggestion
- statistical significance — usually represented as –log₁₀(adjusted p-value) or –log₁₀(unadjusted p-value) 
```

Author: Thanks for catching this! It was left from a previous version - it should be just p-value. Fixed here: 3022f99dbd7ede83ddc2959ecf0c5f1244235dd4.

> By combining these two axes, volcano plots highlight genes that show both large expression changes and strong statistical support, making it easy to identify biologically meaningful up‑ and down‑regulated genes within each cell type.

Reviewer: For the sake of being extra clear, you could update this a bit to reflect that DE genes aren't necessarily biologically meaningful, but a good starting e.g.

Also, since the module is using unadjusted p-values I would be careful not to say that the statistical significance is strongly supported.

```suggestion
By combining these two axes, volcano plots highlight genes that show both large expression changes and statistical support, making it easier to identify potentially biologically meaningful up‑ and down‑regulated genes within each cell type.
```

## 🔍 Interpretation Guidelines

**Up-regulated genes (red-right side)**

Genes plotted in red on the right side represent transcriptional programs increased or activated in the condition for that specific cell type.

> - log2FC > 0 (higher expression in Treatment vs Control)

Reviewer: I think the explanation here might be a bit too specific as we don't necessarily know what conditions will be compared. And also, the comparison is not exactly `condition A vs condition B`, it is `condition A in cell type A` vs all other cells/conditions. Maybe something more general would be a bit more accurate, e.g.

```suggestion
- log2FC > 0 (higher expression in the given cell type or the condition within the given cell type)
```

- adjusted p-value ≤ α (statistically significant)

These genes may indicate:

> - pathway activation

Reviewer: ```suggestion
Depending on the DE comparison, these genes may indicate:

- cell-type-specific markers
- pathway activation
```

- treatment-induced responses
- cell-state transitions

> - altered signaling or regulatory activity

Reviewer: ```suggestion
Depending on the DE comparison, these genes may indicate:

- cell-type-specific markers
- pathway activation
- treatment-induced responses
- cell-state transitions
- altered signaling or regulatory activity
```

**Down-regulated genes (red-left side)**

Genes plotted in red on the left side represent transcriptional programs decreased or inactivated in the condition for that specific cell type.

- log2FC < 0 (lower expression in Treatment vs Control)
- adjusted p-value ≤ α

These genes may represent:

- suppressed pathways
- inhibited regulatory programs
- decreased cell-type–specific functions
- loss of activity due to treatment or condition

**Non-significant genes (grey)**

Genes shown in grey are:

- not significant at the chosen adjusted p-value threshold (p_adj > α), and/or
- show only minor fold-change that does not meet the log2FC threshold.

These genes essentially represent background variation or small changes that are unlikely to be biologically meaningful under current criteria.

## 📌 Summary

> A volcano plot provides a fast and intuitive overview of differential‑expression (DE) patterns within each cell type.

Reviewer: ```suggestion
A volcano plot provides a fast and intuitive overview of differential‑expression (DE) patterns within each cell type and condition.
```

### Color Guide

| **Color** | **Meaning**              | **Criteria**                                       |   **Volcano side**                                              |
|-----------|--------------------------|----------------------------------------------------|-----------------------------------------------------------------|
| **Red**   | Up‑regulated             | log2FC > 0 & adjusted p-value ≤ α                  |   Right side (positive log₂FC)                                  |
| **Red**   | Down‑regulated           | log2FC < 0 & adjusted p-value ≤ α                  |   Left side (negative log₂FC)                                   |
| **Blue**  | Significant p-value only | adjusted p‑value ≤ α but log₂FC < threshold        |   log₂FC; Both sides (near 0 log₂FC)                            |
| **Green** | High fold‑change only    | log₂FC ≥ threshold but adjusted p‑value > α        |   Left = large negative FC; Right = large positive FC           |
| **Grey**  | Not significant          | adjusted p-value > α and log₂FC < threshold        |   Center region around log₂FC ≈ 0                               |

> This visualization helps identify genes that show both strong statistical significance and biologically meaningful fold‑change in a cell‑type–specific manner.

Author: See 3022f99dbd7ede83ddc2959ecf0c5f1244235dd4.

markers_file_condition_value1 <- file.path(markers_dir_module, glue::glue("Res_{resolution_list}_Markers_all_condition_value1.tsv"))
markers_file_condition_value2 <- file.path(markers_dir_module, glue::glue("Res_{resolution_list}_Markers_all_condition_value2.tsv"))
markers_file_condition_value3 <- file.path(markers_dir_module, glue::glue("Res_{resolution_list}_Markers_all_condition_value3.tsv"))

# Create plots directory
plots_dir <- file.path(module_plots_dir, "02-volcano-plots") 
if (!dir.exists(plots_dir)) {
  dir.create(plots_dir)}

plots_dir_overall <- file.path(plots_dir, "01-volcano-plots-overall") 
if (!dir.exists(plots_dir_overall)) {
  dir.create(plots_dir_overall)}

plots_dir_condition1 <- file.path(plots_dir, "02-volcano-plots-condition1") 
if (!dir.exists(plots_dir_condition1)) {
  dir.create(plots_dir_condition1)}

# Source functions
source(paste0(analysis_dir, "/util/function-create-volcano-plot.R"))
```

```{r echo=FALSE, warning=FALSE}
opts_chunk$set(fig.align='center',
               external=TRUE,
               echo=FALSE,
               warning=FALSE,
               fig.pos='H')
a4width <- 8.3
a4height <- 11.7
```

# Volcano Plots Per Cell Type

> The analysis focuses on the top `r n_value` genes ranked by significance to highlight the most biologically relevant up‑ and down‑regulated features.

Reviewer: ```suggestion
The analysis labels the top `r n_value` genes ranked by significance to highlight the most potentially biologically relevant up‑ and down‑regulated features.
```
